src/main/scala/mesosphere/marathon/core/storage/simple/SimplePersistenceStore.scala

```diff
@@ -0,0 +1,159 @@
+  def metric[T](counter: Counter, times: Long = 1): Flow[T, T, NotUsed] = Flow[T].map{ e => counter.increment(times); e}
+  // format: ON
+
+  override def create: Flow[Node, String, NotUsed] =
+    Flow[Node]
+      .via(logNode("Creating a node at "))
+      .via(metric(createMetric))
+      // `groupBy(path.hashCode % parallelism)` makes sure that updates to the same path always land in the same
+      // sub-stream, thus keeping the order of writes to the same path, even with parallelism > 1
+      .groupBy(parallelism, node => Math.abs(node.path.hashCode) % parallelism)
+      .map(node => factory
+        .create()
+        .forPath(node.path, node.data.toArray)
+        .toScala)
+      .mapAsync(parallelism)(identity)
+      .mergeSubstreams
+
+  override def read: Flow[String, Try[Node], NotUsed] =
+    Flow[String]
+      .via(logPath("Reading a node at "))
+      .via(metric(readMetric))
+      .map { path =>
+        factory
+          .getData()
+          .forPath(path)
+          .toScala
+          .map(bytes => Try(Node(path, ByteString(bytes))))
+          .recover {
+            case e: NoNodeException => Failure(e) // re-throw exception in all other cases
+          }
+      }
+      .mapAsync(parallelism)(identity)
+
+  override def update: Flow[Node, String, NotUsed] =
+    Flow[Node]
+      .via(logNode("Updating a node at "))
+      .via(metric(updatedMetric))
+      // `groupBy(path.hashCode % parallelism)` makes sure that updates to the same path always land in the same
+      // sub-stream, thus keeping the order of writes to the same path, even with parallelism > 1
+      .groupBy(parallelism, node => Math.abs(node.path.hashCode) % parallelism)
+      .map(node => factory
+        .setData()
+        .forPath(node.path, node.data.toArray)
+        .toScala
+        .map(_ => node.path))
+      .mapAsync(parallelism)(identity)
+      .mergeSubstreams
+
+  override def delete: Flow[String, String, NotUsed] =
+    Flow[String]
+      .via(logPath("Deleting a node at "))
+      .via(metric(deleteMetric))
+      .map(path => factory
+        .delete()
+        .forPath(path)
+        .toScala
+        .map(_ => path)) // Note: A Void is returned by the delete call and it can not be passed to the mapAsync stage
+      .mapAsync(parallelism)(identity)
+
+  override def children(path: String): Future[Seq[String]] = {
+    childrenMetric.increment()
+    logger.debug(s"Getting children at $path")
+    factory
+      .children()
+      .forPath(path).toScala
+      .map(JavaConverters.asScalaBuffer(_).to[Seq])
+  }
+
+  override def exists(path: String): Future[Boolean] = {
+    existsMetric.increment()
+    logger.debug(s"Checking node existence for $path")
+    factory
+      .checkExists()
+      .forPath(path).toScala
+      .map(stat => if (stat == null) false else true)
+  }
+
+  override def sync(path: String): Future[Done] = {
```

Comment:
Should we have a metric for `sync` too?

Reply:
Also, can we have `def sync(): Future[Done]` too? Because in ZK `path` is not actually used, and it syncs always on `/`.

Reply:
It's not quite correct - we just always call sync with `"/"` for path [here](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/core/storage/store/impl/zk/ZkPersistenceStore.scala#L333). But I agree, that this is a sensible default - I will add it.

Reply:
It is absolutely correct. ZK not matter what path is given when executing `sync` op, syncs on `/`. It is how it works under the hood in ZK.

Reply:
I did not know that :) But in the light of possible future implementation changes on the ZK side, I'm inclined to leave it with a `"/"` as the default.